test/services/sorting/post_sorting_service_test.dart

```diff
@@ -0,0 +1,178 @@
+
+  final postGenerator = FirestorePostGenerator();
+  final positions = GeoPointGenerator.generatePositions(userPosition0, 10, 0);
+  final posts = postGenerator.generatePostsAtDifferentLocations(positions);
+
+  group("Score functions are meaningful", () {
+    double dayDifference(PostFirestore referencePost, PostFirestore post) {
+      final referenceDate = referencePost.data.publicationTime.toDate();
+      final date = post.data.publicationTime.toDate();
+
+      // Need to get the result in miliseconds to have a good precision
+      return referenceDate.difference(date).inMilliseconds /
+          (1000.0 * 60.0 * 60.0 * 24.0);
+    }
+
+    double upvoteDifference(PostFirestore referencePost, PostFirestore post) {
+      final referenceScore = referencePost.data.voteScore;
+      final score = post.data.voteScore;
+
+      return (referenceScore - score).toDouble();
+    }
+
+    double positionDifference(PostFirestore referencePost, PostFirestore post) {
+      final referencePosition = referencePost.location.geoPoint;
+      final position = post.location.geoPoint;
+
+      return Geolocator.distanceBetween(
+        referencePosition.latitude,
+        referencePosition.longitude,
+        position.latitude,
+        position.longitude,
+      );
+    }
+
+    void testOption(
+      PostSortOption option,
+      String expectedBehaviourStr,
+      double Function(PostFirestore, PostFirestore) expectedBehaviour, {
+      bool forcePositiveScore = false,
+    }) {
+      test(
+        "${option.name} score $expectedBehaviourStr",
+        () {
+          final referencePost = postGenerator.generatePostAt(userPosition0);
+          final referenceScore =
+              option.scoreFunction(referencePost, userPosition0);
+
+          final actualScoreDifference = posts
+              .map(
+                (post) =>
+                    referenceScore - option.scoreFunction(post, userPosition0),
+              )
+              .map(
+                (difference) =>
+                    forcePositiveScore ? difference.abs() : difference,
+              );
+
+          final expectedScoreDifference = posts.map(
+            (post) => expectedBehaviour(referencePost, post),
+          );
+          expect(actualScoreDifference, expectedScoreDifference);
+        },
+      );
+    }
+
+    testOption(
+      PostSortOption.hottest,
+      "increases by 1 every upvote, decreases by 1 every day",
+      (referencePost, post) =>
+          upvoteDifference(referencePost, post) -
+          dayDifference(referencePost, post),
+    );
+
+    testOption(
+      PostSortOption.latest,
+      "increases by 1 every every day",
+      dayDifference,
+    );
+
+    testOption(
+      PostSortOption.nearest,
+      "decreases by 1 every meter",
+      positionDifference,
+      forcePositiveScore: true,
+    );
+
+    testOption(
+      PostSortOption.top,
+      "increases by 1 every upvote",
+      upvoteDifference,
+    );
+  });
```

Comment:
Fix in bb5caded1305cb7d8977074298e655dc025ffacb. Tell me if this is clearer.

Reply:
Much clearer; thank you !

Reply:
Much clearer for the `testOption` function, thank you. However, I think that the inner mechanism of the function needs a little more explanation. Like, why `referenceScore` and what it represents, because this function is really abstract. :smile

Reply:
Oh, I understand better. Is 1ffcb65fdd81fcb829228ecd467bf2591ba02662 good for you?

Reply:
Much clearer 💯 . Thank you.